### PR TITLE
Fix crash when carve size is stored as string

### LINKED ISSUE #7756: in debug builds: file carve table broken after size updated
# Bug report

Hi there, when using a debug build there is a type assertion error on loading carve rows after the carver has updated the size column with the actual carve size since the update row function sets the size as a string. I have a fix working locally, but am submitting this and then will attach a PR (and do the CLA).

### What operating system and version are you using?

```
 version = 12.5.1
   build = 21G83
platform = darwin
```

### What version of osquery are you using?

```
version = 5.5.0-1-g41f4aba19
```

### What steps did you take to reproduce the issue?

I setup a local osquery agent with a locally running server in order to build out our file carving server implementation. After the first succesful carve I was no longer able to query the carves table and instead got the following assertion error. `Assertion failed: (data_.f.flags & kIntFlag), function GetInt, file document.h, line 1767..`

### What did you expect to see?

A successful result like 

```
osquery> select * from carves;
+------------+------------------------------------------------------------------+------+------------+-----------+--------------------------------------+--------------------------------------+-------+
| time       | sha256                                                           | size | path       | status    | carve_guid                           | request_id                           | carve |
+------------+------------------------------------------------------------------+------+------------+-----------+--------------------------------------+--------------------------------------+-------+
| 1661380470 | 4132ec7a31f82ed0dbf7d0b30c76b0403cdaec527fc35054e7094e9ba2e7d431 | 2048 | /etc/hosts | SUCCESS   | 573aeba7-c336-40d6-aecd-1c20ea98b256 | 7fc27abf-5506-4b26-8ac9-29681a3c2884 | 0     |
+------------+------------------------------------------------------------------+------+------------+-----------+--------------------------------------+--------------------------------------+-------+
```

### What did you see instead?

```
% osquery/osqueryi --flagfile ~/Desktop/osquery-dev/osquery.local.flags
Using a virtual database. Need help, type '.help'
osquery> select * from carves;
Assertion failed: (data_.f.flags & kIntFlag), function GetInt, file document.h, line 1767.
```

### PR DESCRIPTION
**Pull request by zwass**
In some instances, it was observed that the size is stored as a JSON string rather than number, resulting in an exception if GetInt was called. Check for the type for backwards compatibility.

Fixes #7756.

### CODE REVIEW

**Review by directionless — APPROVED** [review]

File: osquery/tables/forensic/carves.cpp [hunk 1]
```diff
@@ -54,7 +54,14 @@ void enumerateCarves(QueryData& results, const std::string& new_guid) {
     }
 
     if (tree.doc().HasMember("size")) {
-      r["size"] = INTEGER(tree.doc()["size"].GetInt());
+      // In some instances, it was observed that the size is stored as a JSON
+      // string rather than number, resulting in an exception if GetInt was
+      // called. Check for the type for backwards compatibility.
+      if (tree.doc()["size"].IsInt()) {
```
Comment (directionless) [inline osquery/tables/forensic/carves.cpp:60]: Does it make sense to try to track down the inconsistency, and pick one?

**Comment by Smjert** [comment]
@directionless @zwass there was this PR https://github.com/osquery/osquery/pull/7766 (albeit still in draft) that was attempting to fix this.
I left a comment there regarding where to look at the reason for the inconsistency.

**Comment by zwass** [comment]
Ah thank you for pointing that out! I'll take a further look next week.

**Comment by zwass** [comment]
@Smjert @directionless I updated the rest of the code to store the size as a JSON number rather than string. This involved making the `updateCarveValue` function a template function so that it could add an `int` type rather than `string`.

**Review by directionless — APPROVED** [review]

**Pull request merged**